apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/HttpServletInstrumentation.java

```diff
@@ -0,0 +1,84 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.servlet;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import static co.elastic.apm.agent.servlet.ServletInstrumentation.SERVLET_API;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.nameContains;
+import static net.bytebuddy.matcher.ElementMatchers.nameContainsIgnoreCase;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+/**
+ * Instruments HTTP servlets to create transactions.
+ * <p>
+ * If the transaction has already been recorded with the help of {@link FilterChainInstrumentation},
+ * it does not record the transaction again.
+ * But if there is no filter registered for a servlet,
+ * this makes sure to record a transaction in that case.
+ * </p>
+ */
+public class HttpServletInstrumentation extends ElasticApmInstrumentation {
```

Comment:
Why do we need `HttpServletInstrumentation` and `ServletInstrumentation`?

Reply:
Wasn't this what you requested?

Reply:
Are you referring to https://github.com/elastic/apm-agent-java/pull/422#discussion_r248605696? Now, `HttpServlet`s are still instrumented twice, aren't they? I think we only need the `ServletInstrumentation`.

Reply:
Right, always calls from it, removing .....